SpatialGDK/Source/SpatialGDK/Public/Interop/CreateEntityHandler.h

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+
+#include "CoreMinimal.h"
+#include "Interop/SpatialOSDispatcherInterface.h"
+#include "SpatialCommonTypes.h"
+#include "SpatialConstants.h"
+
+#include <WorkerSDK/improbable/c_schema.h>
+#include <WorkerSDK/improbable/c_worker.h>
+
+DECLARE_CYCLE_STAT(TEXT("CreateEntityHandler"), STAT_CreateEntityHandler, STATGROUP_SpatialNet);
+
+namespace SpatialGDK
+{
+class CreateEntityHandler
+{
+	DECLARE_LOG_CATEGORY_CLASS(LogCreateEntityHandler, Log, All);
+
+public:
+	void AddRequest(Worker_RequestId RequestId, CreateEntityDelegate Handler)
+	{
+		check(Handler.IsBound());
+		Handlers.Add(RequestId, Handler);
+	}
+	void ProcessOps(const TArray<Worker_Op>& Ops)
```

Comment:
The process ops functions across the different handlers seem like similar code- could they be combined into a class which the different handlers inherit from? Maybe templated by op type/handler

Reply:
The only issue I see with this is that we'll have to use pointers to members to truly templatize this - so it'll be somewhat verbose:

```
template <uint8 OpTypeId, class TOpType, TOpType Worker_OpUnion::* TypedOpPtr>
class OpHandlerBase
{
	void ProcessOps(const TArray<Worker_Op>& Ops)
	{
		for (const Worker_Op& Op : Ops)
		{
			if (Op.op_type == OpTypeId)
			{
				const TOpType& TypedOp = Op.*TypedOpPtr;
				ProcessOp(Op, TypedOp);
			}
		}
	}
};

class CreateEntityHandler : public OpHandlerBase<WORKER_OP_TYPE_CREATE_ENTITY_RESPONSE, 
Worker_CreateEntityResponseOp, &Worker_OpUnion::create_entity_response>
{
...
}
```

Reply:
yep- probably not worth doing. Appreciate you taking the time to consider the suggestion considering it doesn't make all that much sense 😄